app/src/main/java/ch/epfl/polybazaar/UI/SubmitOffer.java

```diff
@@ -0,0 +1,99 @@
+        setContentView(R.layout.activity_submit_offer);
+        inputOffer = findViewById(R.id.offer);
+        listing = (Listing)getIntent().getSerializableExtra(LISTING);
+    }
+
+    public void submitOffer(View view) {
+        if (!String.valueOf(inputOffer.getText()).isEmpty()) {
+            double offer = Double.parseDouble(inputOffer.getText().toString());
+            sendOffer(offer, listing, SubmitOffer.this);
+            finish();
+        } else {
+            Toast.makeText(getApplicationContext(), R.string.offer_invalid, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    public void cancelOfferMaking(View view) {
+        finish();
+    }
+
+    /**
+     * sends an offer
+     * @param offer the offer amount
+     */
+    public static void sendOffer(Double offer, Listing listing, Activity activity) {
+        String senderEmail = AuthenticatorFactory.getDependency().getCurrentUser().getEmail();
+        ChatMessage message = new ChatMessage(senderEmail, listing.getUserEmail(),
+                listing.getId(),
+                ChatMessage.OFFER_MADE + offer.toString(),
+                new Timestamp(new Date(System.currentTimeMillis())));
+        final String newMessageID = randomUUID().toString();
+        message.setId(newMessageID);
+        message.save().addOnSuccessListener(aVoid ->
+                Toast.makeText(activity.getApplicationContext(), R.string.offer_sent, Toast.LENGTH_LONG).show());
+    }
+
+    /**
+     * Accepts or refuses the offer
+     * @param offer the offer amount
+     * @param receivedOfferMessage the message received by the seller asking to accept or refuse the offer
+     * @param offerStatus OFFER_ACCEPTED or OFFER_REFUSED
+     */
+    public static void processOffer(Double offer, ChatMessage receivedOfferMessage, String offerStatus) {
+        String listingID = receivedOfferMessage.getListingID();
+        String sellerEmail = receivedOfferMessage.getReceiver();
+        String buyerEmail = receivedOfferMessage.getSender();
+        String messageContent = OFFER_PROCESSED + offerStatus + offer.toString();
+        ChatMessage message = new ChatMessage(sellerEmail, buyerEmail,
+                listingID,
+                messageContent,
+                new Timestamp(new Date(System.currentTimeMillis())));
+        final String newMessageID = randomUUID().toString();
+        message.setId(newMessageID);
+        message.save().addOnSuccessListener(aVoid -> receivedOfferMessage.delete());
+        // disable offer making and put sold banner
+        if (offerStatus.equals(OFFER_ACCEPTED)) {
+            Listing.updateField(Listing.LISTING_ACTIVE, message.getListingID(), false);
+            Listing.updateField(Listing.PRICE, message.getListingID(), Listing.SOLD);
+            Map<String, Object> map =new HashMap<>();
+            map.put(LiteListing.PRICE, LiteListing.SOLD);
+            LiteListing.updateMultipleFields(message.getListingID(), map);

```

Comment:
Using 2 updateField for 2 fields is fine, but you should also use updateField for the LiteListing. If the static method isn't available in LiteListing, you can easily add it, it will be pretty much a copy paste from Listing.updateField

Reply:
Done